stdlib/public/Concurrency/Task.cpp

```diff
@@ -287,6 +298,31 @@ AsyncTaskAndContext swift::swift_task_create_future_f(
     futureContext->indirectResult = futureFragment->getStoragePtr();
   }
 
+  // Perform additional linking between parent and child task.
+  if (parent) {
+    // FIXME: can we allocate this in the task local allocator?
+    // TODO: Do we need to delete it manually or is the record infra handling this?
+    if (flags.task_isGroupChildTask()) {
+      // this task is a child of a task group
+      fprintf(stderr, "[%s:%d] (%s): create group child task record in parent [%d] to %d [group:%d]\n", __FILE__, __LINE__, __FUNCTION__, parent, task, group);
+      swift_task_addStatusRecord(parent,
+                                 new GroupChildTaskStatusRecord(group, task));
+    } else {
+      // just a normal child task
+      fprintf(stderr, "[%s:%d] (%s): create child task record in parent [%d] to %d %d\n", __FILE__, __LINE__, __FUNCTION__, parent, task);
+      swift_task_addStatusRecord(parent,
+                                 new ChildTaskStatusRecord(task));
+    }
```

Comment:
not sure if this is the best way to do this, but gets the job done -- when we cancel the task we then iterate over *all* children, or if we cancelAll the group we iterate over all _group (only)_ children